crates/re_data_ui/src/editors.rs

```diff
@@ -257,28 +257,48 @@ fn edit_marker_shape_ui(
 
     egui::ComboBox::from_id_source("marker_shape")
         .selected_text(marker_text) // TODO(emilk): Show marker shape in the selected text
-        .width(100.0)
+        .width(ui.available_width().at_most(100.0))
         .height(320.0)
         .show_ui(ui, |ui| {
             // no spacing between list items
             ui.spacing_mut().item_spacing.y = 0.0;
 
-            // Hack needed for ListItem to click its highlight bg rect correctly:
-            ui.set_clip_rect(
-                ui.clip_rect()
-                    .with_max_x(ui.max_rect().max.x + ui.spacing().menu_margin.right),
+            let item_width = 100.0;
+
+            // workaround to force `ui.max_rect()` to reflect the content size
+            ui.allocate_space(egui::vec2(item_width, 0.0));
+
+            let background_x_range = ui
+                .spacing()
+                .menu_margin
+                .expand_rect(ui.max_rect())
+                .x_range();
+
+            re_ui::list_item2::list_item_scope(
+                ui,
+                "marker_shape",
+                Some(background_x_range),
+                |ui| {
+                    for marker in MarkerShape::ALL {
+                        let response = ctx
+                            .re_ui
+                            .list_item2()
+                            .selected(edit_marker == marker)
+                            .show_flat(
+                                ui,
+                                re_ui::list_item2::LabelContent::new(marker.to_string())
+                                    .min_desired_width(item_width)
+                                    .with_icon_fn(|_re_ui, ui, rect, visuals| {
+                                        paint_marker(ui, marker.into(), rect, visuals.text_color());
+                                    }),
+                            );
```

Comment:
I can't immediately think of one.

One area to think about is to have some `DataUi` like thing that would make our data objects somehow capable of producing `impl ListItemContent`. If `MarkerShape` had this behaviour, it could be passed directly (or as `marker.as_content()` or sth) to  `.show_flat`. I think we do need some hindsight to be able to design such a system, though.